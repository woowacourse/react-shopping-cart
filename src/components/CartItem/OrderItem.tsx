import * as styles from './CartItem.style';
import { CartItemType } from '../../types/cartItem';

interface OrderItemProps {
  item: CartItemType;
}

function OrderItem({ item }: OrderItemProps) {
  const { id: cartItemId, product, quantity: cartQuantity } = item;
  const { name, price, imageUrl } = product;

  return (
    <div key={cartItemId} css={styles.cartItemFrameCss}>
      <div css={styles.cartItemInfoCss}>
        <img
          css={styles.cartItemImgCss}
          src={imageUrl || './assets/default.png'}
          alt={name}
          onError={(e) => {
            e.currentTarget.src = './assets/default.png';
          }}
        />
        <div css={styles.cartItemInfoTextCss}>
          <div>
            <p css={styles.cartItemQuantityCss}>{name}</p>
            <p css={styles.cartItemPriceCss}>{(price * cartQuantity).toLocaleString()}원</p>
          </div>
          <p css={styles.cartItemQuantityCss}>{cartQuantity}개</p>
        </div>
      </div>
    </div>
  );
}

export default OrderItem;
