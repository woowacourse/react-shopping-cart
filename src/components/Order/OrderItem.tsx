import { CartProduct } from '../../types/cart';
import { woowaLogo } from '../../assets/index';
import styled from '@emotion/styled';

interface CartItemProps {
  cartItem: CartProduct;
}
function OrderItem({ cartItem }: CartItemProps) {
  return (
    <CartItemContainer>
      <ProductRow>
        <CartProductImage
          src={cartItem.product.imageUrl}
          onError={(e) => {
            const target = e.target as HTMLImageElement;
            target.src = woowaLogo;
          }}
        />
        <CartContent>
          <ProductTitle>{cartItem.product.name}</ProductTitle>
          <ProductPrice>{cartItem.product.price.toLocaleString()}원</ProductPrice>
          <StepperContainer>
            <StepperQuantity data-testid="item-quantity">{cartItem.quantity}개</StepperQuantity>
          </StepperContainer>
        </CartContent>
      </ProductRow>
    </CartItemContainer>
  );
}
export default OrderItem;

export const ProductTitle = styled.h2`
  font-weight: 700;
  font-size: 16px;
  line-height: 100%;
`;

export const ProductPrice = styled.p`
  margin: 4px 0 0 0;
  font-weight: 500;
  font-size: 12px;
  line-height: 15px;
  vertical-align: middle;
`;

export const StepperContainer = styled.div`
  display: flex;
  align-items: center;
  margin-top: 8px;
`;

export const StepperButton = styled.button`
  width: 24px;
  height: 24px;
  border: 1px solid #ddd;
  border-radius: 6px;
  background-color: white;
  font-size: 18px;
  font-weight: 500;
  color: #666;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: all 0.2s ease;

  &:hover {
    background-color: #f5f5f5;
    border-color: #ccc;
  }

  &:active {
    background-color: #e8e8e8;
  }

  &:disabled {
    opacity: 0.4;
    cursor: not-allowed;
    &:hover {
      background-color: white;
      border-color: #ddd;
    }
  }
`;

export const StepperQuantity = styled.span`
  text-align: center;
  font-size: 12px;
  font-weight: 500;
  color: #0a0d13;
`;

export const CartItemContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  padding: 16px 0;
  border-top: 1px solid rgba(0, 0, 0, 0.1);
`;

export const CartProductImage = styled.img`
  width: 80px;
  height: 80px;
  border-radius: 8px;
  object-fit: cover;
  background: #eee;
`;

export const ProductRow = styled.div`
  display: flex;
  align-items: flex-start;
  gap: 16px;
`;

export const CartContent = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  gap: 4px;
  flex: 1;
`;
