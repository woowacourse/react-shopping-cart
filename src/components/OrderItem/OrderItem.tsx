import { CartItemType } from "../../types/response";
import {
  CountContainer,
  ItemContainer,
  ItemContent,
  ItemDetail,
  ItemInfo,
  ItemPrice,
  ItemTitle,
  ProductImage,
} from "./OrderItem.styles";

interface OrderItemProps {
  orderItem: CartItemType;
}

function OrderItem({ orderItem }: OrderItemProps) {
  const { product, quantity } = orderItem;

  return (
    <>
      <li css={ItemContainer}>
        <div css={ItemInfo}>
          <img
            css={ProductImage}
            src={product.imageUrl}
            onError={(error) => {
              error.currentTarget.src = "default-cartItem.png";
            }}
          ></img>
          <div css={ItemContent}>
            <div css={ItemDetail}>
              <p css={ItemTitle}>{product.name}</p>
              <p css={ItemPrice}>{product.price.toLocaleString()}원</p>
            </div>
            <div css={CountContainer}>
              <p>{quantity}개</p>
            </div>
          </div>
        </div>
      </li>
    </>
  );
}

export default OrderItem;
