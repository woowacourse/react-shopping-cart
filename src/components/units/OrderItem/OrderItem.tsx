import React from 'react';
import Styled from './OrderItem.styles';
import OrderItemImageURL from '../../../assets/images/brave.png';

type OrderItemProps = {
  title: string;
  quantity: number;
  imageUrl?: string;
};

const OrderItem = (props: OrderItemProps) => {
  const { title, imageUrl, quantity } = props;

  return (
    <Styled.Root>
      <Styled.Image src={imageUrl} alt={title} />
      <Styled.Info>
        <Styled.Title>{title}</Styled.Title>
        <Styled.Quantity>수량 : {quantity}개</Styled.Quantity>
      </Styled.Info>
    </Styled.Root>
  );
};

OrderItem.defaultProps = {
  imageUrl: OrderItemImageURL,
};

export default OrderItem;
