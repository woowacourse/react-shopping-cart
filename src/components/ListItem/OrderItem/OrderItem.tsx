import ListItem from '../ListItem';
import ListItemBody from '../ListItemBody';
import ListItemInfo from '../ListItemInfo';
import Text from '../../@common/Text/Text';

import { CartItemProps } from '../../../types/cartItem';
import { OrderItemQuantityStyle } from './OrderItem.styles';

function OrderItem({ cartItem }: { cartItem: CartItemProps }) {
  return (
    <ListItem>
      <ListItemBody
        imageUrl={cartItem.product.imageUrl}
        name={cartItem.product.name}
      >
        <ListItemInfo
          name={cartItem.product.name}
          price={cartItem.product.price}
        >
          <div css={OrderItemQuantityStyle}>
            <Text variant="caption">{cartItem.quantity}ê°œ</Text>
          </div>
        </ListItemInfo>
      </ListItemBody>
    </ListItem>
  );
}

export default OrderItem;
