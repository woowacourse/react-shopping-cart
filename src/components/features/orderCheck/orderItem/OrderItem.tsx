import { isValidImageUrl } from '@/shared';
import { CartItemType } from '../../cart/types';
import * as S from './OrderItem.styles';
import defaultImage from '/assets/default_product.png';
import { Separator } from '@/components/common';

interface OrderItemProps {
  cartItem: CartItemType;
}

function OrderItem({ cartItem }: OrderItemProps) {
  return (
    <S.Container>
      <Separator />
      <S.InfoContainer>
        <S.PreviewBox>
          <S.PreviewImage
            src={
              isValidImageUrl(cartItem.product.imageUrl)
                ? cartItem.product.imageUrl
                : defaultImage
            }
            alt="상품 이미지"
          />
        </S.PreviewBox>
        <S.InfoBox>
          <S.CartProductInfo>
            <S.CartProductTitle>{cartItem.product.name}</S.CartProductTitle>
            <S.CartProductPrice>
              {`${(
                cartItem.product.price * cartItem.quantity
              ).toLocaleString()}원`}
            </S.CartProductPrice>
          </S.CartProductInfo>
          <S.Text>{cartItem.quantity}개</S.Text>
        </S.InfoBox>
      </S.InfoContainer>
    </S.Container>
  );
}

export default OrderItem;
