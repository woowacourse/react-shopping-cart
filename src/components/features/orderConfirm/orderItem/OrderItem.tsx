import { isValidImageUrl } from '../../../../utils/isValidImageUrl';
import FallbackImage from '../../../common/fallbackImage/FallbackImage';
import Separator from '../../../common/separator/Separator';
import { CartItemType } from '../../cart/types';
import * as S from './OrderItem.styles';

interface OrderItemProps extends CartItemType {}

function OrderItem({ id, quantity, product }: OrderItemProps) {
  return (
    <S.Container data-testid={`OrderItem-${id}`}>
      <Separator />
      <S.InfoContainer>
        <S.PreviewBox>
          {isValidImageUrl(product.imageUrl) ? (
            <S.PreviewImage src={product.imageUrl} alt="상품 이미지" />
          ) : (
            <FallbackImage />
          )}
        </S.PreviewBox>
        <S.InfoBox>
          <S.CartProductInfo>
            <S.CartProductTitle>{product.name}</S.CartProductTitle>
            <S.CartProductPrice>
              {`${(product.price * quantity).toLocaleString()}원`}
            </S.CartProductPrice>
          </S.CartProductInfo>
          <S.Text>{quantity}개</S.Text>
        </S.InfoBox>
      </S.InfoContainer>
    </S.Container>
  );
}

export default OrderItem;
