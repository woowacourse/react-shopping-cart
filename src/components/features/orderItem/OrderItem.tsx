import * as S from './OrderItem.styles';
import type { CartItemType } from '../../../types/response';

interface OrderItemProps {
  cartData: CartItemType;
}

const OrderItem = ({ cartData }: OrderItemProps) => {
  return (
    <div css={S.orderItemWrapper}>
      <div css={S.orderItemStyle}>
        <img src={cartData.product.imageUrl} alt={cartData.product.name} />
        <div css={S.orderInfoStyle}>
          <h3 css={S.orderItemNameStyle}>{cartData.product.name}</h3>
          <p css={S.orderItemPriceStyle}>
            {cartData.product.price.toLocaleString()}원
          </p>
          <p css={S.orderItemQuantityStyle}>{cartData.quantity}개</p>
        </div>
      </div>
    </div>
  );
};

export default OrderItem;
