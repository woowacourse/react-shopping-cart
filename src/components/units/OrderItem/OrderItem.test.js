import React from 'react';
import { render } from '../../../test-utils';
import OrderItem from './OrderItem';

describe('OrderItem 테스트', () => {
  test('matches snapshot', () => {
    const title = '지그의 다쿠아즈';
    const utils = render(<OrderItem title={title} quantity="5" />);

    expect(utils.container).toMatchSnapshot();
  });
});
