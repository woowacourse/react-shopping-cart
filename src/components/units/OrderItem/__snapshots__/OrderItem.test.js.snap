// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderItem 테스트 matches snapshot 1`] = `
<div>
  <li
    class="css-c6rm5t-Root e1j29uye4"
  >
    <img
      alt="item-image"
      class="css-lbmuvq-Image e1j29uye3"
      src="brave.png"
    />
    <div
      class="css-1vf5l7v-Info e1j29uye2"
    >
      <div
        class="css-1nsby4x-Title e1j29uye1"
      >
        지그의 다쿠아즈
      </div>
      <div
        class="css-1pci65m-Quantity e1j29uye0"
      >
        수량 : 
        5
        개
      </div>
    </div>
  </li>
</div>
`;
